### PR TITLE
Chore: convert `MetricSelect` test to RTL

### LINKED ISSUE #56800: Convert `public/app/core/components/Select/MetricSelect.test.tsx` to RTL

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/56800

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/Select/MetricSelect.test.tsx [hunk 1]
```diff
@@ -1,50 +1,52 @@
...
-        isMulti: false,
-        isClearable: false,
-        backspaceRemovesValue: false,
-        isSearchable: true,
-        maxMenuHeight: 500,
-        placeholder: 'Select Reducer',
-      });
+  it('passes the placeholder, options and onChange correctly to Select', async () => {
+    render(<MetricSelect {...props} />);
+    const metricSelect = screen.getByRole<HTMLInputElement>('combobox');
```
Comment (Clarity-89) [inline public/app/core/components/Select/MetricSelect.test.tsx:29]:
```suggestion
    const metricSelect = screen.getByRole('combobox');
```
I don't think providing type param is necessary here? Better infer them whenever possible.

File: public/app/core/components/Select/MetricSelect.test.tsx [hunk 2]
```diff
@@ -1,50 +1,52 @@
...
-      select.props().onChange({ value: 'foo' }, { action: 'select-option', option: undefined });
+  it('has the correct noOptionsMessage', () => {
+    const propsWithoutOptions = {
+      ...props,
+      options: [],
+    };
+    render(<MetricSelect {...propsWithoutOptions} />);
 
-      expect(select.props().noOptionsMessage).toBeDefined();
+    const metricSelect = screen.getByRole<HTMLInputElement>('combobox');
```
Comment (Clarity-89) [inline public/app/core/components/Select/MetricSelect.test.tsx:46]:
```suggestion
    const metricSelect = screen.getByRole('combobox');
```

**Review by Clarity-89 — APPROVED** [review]
Looks good, just a minor comment.

File: public/app/core/components/Select/MetricSelect.test.tsx (hunk 1 above)
Comment (ashharrison90) [reply public/app/core/components/Select/MetricSelect.test.tsx:29]: @Clarity-89 ah yes you're right, it's leftover from when i was playing with the `value` of the input - i'll remove 👍

**Pull request merged**